src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetries.cs

```diff
@@ -30,8 +33,10 @@ internal SecondLevelRetries()
         protected internal override void Setup(FeatureConfigurationContext context)
         {
             var  retryPolicy = GetRetryPolicy(context.Settings);
+            var slrStorage = new SlrStatusStorage();
 
             context.Container.RegisterSingleton(typeof(SecondLevelRetryPolicy), retryPolicy);
+            context.Container.RegisterSingleton(typeof(SlrStatusStorage), slrStorage);
```

Comment:
In general we try to avoid the usage of `IBuilder` as much as possible. Since you already created the `SlrStatusStorage` you can directly pass it to the lambda of the `SecondLevelRetriesBehavior` like you do and try to pass it into the Startup task directly (I know currently not possible but we should/could come up with a way to do it :D) 

Same applies for FLR. We can consider this outside the scope of this PR.